Initial changelog (#102)
jasmin-lang/jasmin: CHANGELOG.md

# Jasmin 22.0

This release is the result of more than two years of active development. It thus
contains a lot of new functionalities compared to Jasmin 21.0, the main ones
being listed below, but also a lot of breaking changes. Please upgrade with
care.

Here are the main changes of the release.
- **A new kind of function, subroutines,** in addition to inline functions and
  export functions. They are declared with `fn` only, while inline functions are
  declared with `inline fn` and export functions with `export fn`. This is a
  breaking change, since before `fn` was a synonym for `inline fn`. Unlike
  inline functions, they are proper functions. Unlike export functions, they are

>   internal. As such, they do not need to respect the calling conventions of C

Reviewer: ```suggestion
  internal. As such, they do not need to respect any standard calling conventions
```

  and are therefore a bit more flexible.

- **New storage modifiers `reg ptr` and `stack ptr`** to declare arrays.

>   `reg ptr` is used to store the address of a stack array in a register. The

Reviewer: ```suggestion
  `reg ptr` is used to store the address of an array in a register. The
```
